app/presenters/change_history_presenter.rb

```diff
@@ -30,6 +30,7 @@ def change_notes
         .joins(:edition)
         .where(editions: { document: document })
         .where("user_facing_version <= ?", version_number)
+        .where.not(public_timestamp: nil)
```

Comment:
I've gone back and forth on my thinking about this - I keep having an inkling that there's a better way to deal with this, but not come to definite conclusions. 

I think basically my concern is whether this will cause any issues for draft content. I know there's currently an issue anyway since these draft documents with un-stamped change notes will fail to be sent in the same way that this test fails, but I'm also wondering whether filtering them out will also cause us issues at some point. 

I guess we couldn't give these timestamps in the same way as we do for published editions, since the edition doesn't have a public timestamp. But wondering whether we could just give them _some_ timestamp, or enforce timestamps on the frontend, or solve in some other way. Or whether the implications are fine?

So more of a discussion point given that you've thought about this a lot, than a definite suggestion.

Reply:
Thanks for giving this some thought. I can understand your concern with this, there is definitely something "sweep this under the carpet" with a query condition like this.

I can explain some of my rationale though. I see the primary concern is having the Publishing API produce a change history that creates a valid schema. So I'm prioritising outputting only valid data over outputting complete data. This seems preferable to me as it holds up Publishing API's side of the contract. Changing the schema also seems a growing bit of work which creates confusing scenarios (change notes without timestamps).

The tricky thing, as you've addressed, is that the state of the item can be valid and not have a timestamp. So we can end up in the draft situation. I think that's ok to not have the change note show before the date is available.

You're probably right that one day an issue will occur. I'd hope it's only some confusion of a draft for someone.

Annoyingly though this change note stuff is niche, because apps like Whitehall send a whole change history in their payload (if anything I think this ChangeNote system is bad but that's a longer story) which I think somewhat explains why this hasn't caused user issues.

Reply:
Sorry for the slow reply @kevindew, and thanks for the explanation.

That makes sense - I think in a perfect world it'd be nice for the schemas to represent reality, so having a draft schema vs published schema. But I can see how this can get confusing, especially since I've been looking at the current state of confusion in schemas! 

So I think this is a pragmatic approach, we can review in future, and is definitely an improvement on now when the draft content with changenotes would error.